spring-integration-core/src/main/java/org/springframework/integration/dsl/CorrelationHandlerSpec.java

```diff
@@ -135,7 +135,7 @@ public S groupTimeout(Function<MessageGroup, Long> groupTimeoutFunction) {
 	 * @see AbstractCorrelatingMessageHandler#setTaskScheduler(TaskScheduler)
 	 */
 	public S taskScheduler(TaskScheduler taskScheduler) {
-		Assert.notNull(taskScheduler, "'taskScheduler' must not be null.");
```

Comment:
I think the assertion is better here (see my previous comment) - someone might delete that one without knowledge of this super() call.